docs/modules/ROOT/pages/design/cloudevents.adoc

```diff
@@ -0,0 +1,125 @@
+* Cloud Events that could not be delivered are replayed to KubeArchive automatically when KubeArchive becomes available
+  again
+
+== Design
+
+KubeArchive needs to use the functionality provided by Knative Eventing, Brokers and Triggers as well as features
+provided by the Cloud Events Specification to achieve an event driven architecture, where events are not lost and events
+can be replayed safely without worries about side effects.
+
+This diagram provides a high level overview of the life cycle of Cloud Events look like in KubeArchive and how the
+different components of the Cloud Event Delivery Pipeline interact with eachother.
+
+image::ce-pipeline.png[]
+
+=== Knative Eventing Brokers
+
+Instead of the ApiServerSource sending Cloud Events directly to the sink, the Cloud Events should instead be sent to a
+Knative Eventing Broker. Brokers can receive, persist, and send messages. Using the broker allows messages to be
+sent somewhere that they can be stored long term and is separate from the KubeArchive sink and Knative Eventing
+ApiServerSource, so if either are restarted, the Cloud Events won't be lost. KubeArchive will use two brokers. One
+primary broker (called broker) and secondary broker (called dead letter sink). The default configuration of
+Knative Eventing will have these brokers be a created as In Memory Brokers, but changing the Knative Eventing
+configuration can allow this broker to be created, for example, as a Kafka Broker instead
+
+[IMPORTANT]
+====
+For Production workloads, Knative Eventing strongly recommends using Brokers types other than the In Memory Broker.
+====
+
+==== Primary Broker
+
+The primary broker receives all of Cloud Events from the ApiServerSources created by the KubeArchive Operator.
+KubeArchive uses a trigger to subscribe to Cloud Events from this broker. The primary broker is configured as follows:
+
+[source,yaml]
+----
+spec:
+  delivery:
+    retry: 4
+    backoffPolicy: exponential
+    # ISO 8601 format
+    backoffDelay: PT0.5S
+    deadLetterSink:
+      ref:
+        apiVersion: eventing.knative.dev/v1
+        kind: Broker
+        name: kubearchive-dls
+----
+
+If the initial delivery fails, the broker will attempt to deliver the cloud event 4 additional times. When attempting to
+retry delivering the event, the broker will use an exponential backoff algorithm to calculate the amount of time to wait
+between retry attempts, and if all retry attempts fail, the broker will send the cloud event to the dead letter sink.
+
+==== Dead Letter Sink
+
+The dead letter sink is a second broker that stores cloud events that could not be delivered to the KubeArchive sink. If
+the sink has received every message from the broker, then there will be no cloud events stored in the dead letter sink.
+If the sink becomes unreachable or unable to process cloud events, the dead letter sink will start to receive cloud
+events from the primary broker.
```

Comment:
```suggestion
The dead letter sink stores the Cloud Events that could not be delivered to the KubeArchive sink. This broker does not contain any message if the connectivity between the Primary Broker and the KubeArchive sink is correct.
```

Reply:
I think this is enough:
```
The dead letter sink stores the Cloud Events that could not
be delivered to the KubeArchive sink.
```

Reply:
I changed the wording in this section according to the suggestions here and added a sentence to mention that KubeArchive can reject a cloud event that it receives and added a link to the section on http error codes